Controllers/Ingredient_ComponentsController.cs

```diff
@@ -104,38 +97,34 @@ public ActionResult Create(int IngredID)
+            Ingredient ingredient = await _recitopiaDbContext.Ingredient.FindAsync(ingredID);
 
             //Assign to temp local to put on view
             ViewBag.IngredName = ingredient.Ingred_name;
-            ViewBag.IngredId = IngredID;
+            ViewBag.IngredId = ingredID;
 
-            ViewBag.Ingred_Id = new SelectList(_recitopiaDbContext.Ingredient.Where(m => m.Customer_Id == CustomerId).OrderBy(m => m.Ingred_name), "Ingredient_Id", "Ingred_name");
+            ViewBag.Ingred_Id = new SelectList( _recitopiaDbContext.Ingredient.Where(m => m.Customer_Id == CustomerId).OrderBy(m => m.Ingred_name), "Ingredient_Id", "Ingred_name");
             ViewBag.Comp_Id = new SelectList(_recitopiaDbContext.Components.Where(m => m.Customer_Id == CustomerId).OrderBy(m => m.Comp_Sort), "Comp_Id", "Component_Name");
             //ViewBag.Comp_Id = new SelectList(db.Components.Select(m => new { m.Comp_Id, m.Component_Name }));
 
 
             //BUILD VIEW FOR ALREADY ADDED COMPONENTS
-            var query =
-               from t1 in _recitopiaDbContext.Ingredient_Components.AsQueryable()
-               join t2 in _recitopiaDbContext.Ingredient.AsQueryable() on t1.Ingred_Id equals t2.Ingredient_Id into t2g
-               from t2 in t2g.DefaultIfEmpty() //DefaultIfEmpty is used for left joins
-               join t3 in _recitopiaDbContext.Components.AsQueryable() on t1.Comp_Id equals t3.Comp_Id
-               where t1.Ingred_Id == IngredID
-               where t1.Customer_Id == CustomerId
-               orderby t3.Component_Name
-               select new View_All_Ingredient_Components()
-               {
-                   Id = t1.Id,
-                   Customer_Id = t1.Customer_Id,
-                   Ingred_Id = t1.Ingred_Id,
-                   Ingred_name = t2.Ingred_name,
-                   Comp_Id = t1.Comp_Id,
-                   Component_Name = t3.Component_Name,
-
-               };
+            var ingredientComponents = await _recitopiaDbContext.Ingredient_Components
+                            .Include(ri => ri.Ingredients)
+                            .Include(ri => ri.Components)
+                            .Where(ri => ri.Customer_Id == CustomerId && ri.Ingred_Id == ingredID)
+                            .Select(ri => new View_All_Ingredient_Components()
+                            {
+                                Id = ri.Id,
+                                Customer_Id = ri.Customer_Id,
+                                Ingred_Id = ri.Ingred_Id,
+                                Ingred_name = ri.Ingredients.Ingred_name,
+                                Comp_Id = ri.Comp_Id,
+                                Component_Name = ri.Components.Component_Name
+                            })
+                            .ToListAsync();
             //Get View and filter and sort            
-            var TempV = query.ToList();
+            var TempV = ingredientComponents;
```

Comment:
what? why? use `ingredientComponents` in the `foreach` loop below.

Reply:
Early code when learning.  Cleaned up.